rust/noosphere/tests/client_to_gateway.rs

```diff
@@ -55,22 +51,24 @@ async fn gateway_tells_you_its_identity() -> Result<()> {
 
     let listener = TcpListener::bind("127.0.0.1:0").unwrap();
     let gateway_address = listener.local_addr().unwrap();
-
-    let gateway_sphere_identity = gateway_workspace.sphere_identity().await.unwrap();
     let client_sphere_identity = client_workspace.sphere_identity().await.unwrap();
-
-    let gateway_sphere_context = gateway_workspace.sphere_context().await.unwrap();
+    let mut gateway_sphere_context = gateway_workspace.sphere_context().await.unwrap();
+    {
+        let mut sphere_context = gateway_sphere_context.sphere_context_mut().await?;
+        sphere_context
+            .db_mut()
+            .set_key(COUNTERPART, client_sphere_identity)
+            .await?;
```

Comment:
In the spirit of these tests, lets use the CLI command implementation e.g., `config_set(ConfigSetCommand::Counterpart { did }, &mut gateway_workspace)`

https://github.com/subconsciousnetwork/noosphere/blob/39a58e6335071662fef989935907025d0732e3cf/rust/noosphere-cli/src/native/commands/sphere/config.rs#L11-L24